clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -101,12 +98,12 @@ public synchronized void subscribe(Collection<String> topics) {
     @Override
     public synchronized void subscribe(Pattern pattern, final ConsumerRebalanceListener listener) {
         ensureNotClosed();
-        committed.clear();
+        mockLogs.values().forEach(log -> log.setCommitted(null));
```

Comment:
To make the mock consumer consistent with the real implementation, it's not clear why we'd want this. Committed offsets persist even after re-subscribing. Do you know if we depend on this logic anywhere?